test/testhelpers/OffsetArrays.jl

```diff
@@ -5,141 +5,278 @@
 # This test file is designed to exercise support for generic indexing,
 # even though offset arrays aren't implemented in Base.
 
-module OffsetArrays
+module OffsetArrays # OffsetArrays@1.0.0 without compat and docstrings
 
-using Base: Indices, IndexCartesian, IndexLinear, tail
+using Base: Indices, tail, @propagate_inbounds
+@static if !isdefined(Base, :IdentityUnitRange)
```

Comment:
This is always defined on `master`, which is what this PR targets.

I didn't go through the rest carefully, is the idea here to basically "sync" the version here with that in OffsetArrays.jl? Seems reasonable.

Reply:
Ooops, it turns out that the first c2fd49e commit is broken here and I've separate this commit out in #37643. To first upgrade the OffsetArrays version.